tests/test_pages.py

```diff
@@ -430,3 +436,151 @@ def nextPage(self):
         assert request.metadata is not None
         assert request.metadata.probability == 1.0
     assert type(page.nextPage) == Request
+
+
+def test_price_selector():
+    class CustomProductPage(ProductPage):
+        @field
+        def price(self):
+            return self.css("div::text")
+
+    html = b"""
+    <!DOCTYPE html>
+    <html>
+        <body>
+            <div>$13.2</div>
+        </body>
+    </html>
+    """
+    url = RequestUrl("https://example.com")
+    page = CustomProductPage(response=HttpResponse(url=url, body=html))
+    assert page.price == "13.2"
+    assert page.currency is None
+    assert page.currencyRaw == "$"
+
+    # access currency fields before the price field
```

Comment:
It may be nice to add a check which ensures the extraction code is not called multiple times; specifically, ensure the user defined code in the .price field is only executed once.

Reply:
It won't be called multiple times only if the user specified cached=True, because `hasattr(self, "price")` calls it again via `__get__` :-/

Reply:
Even if it doesn't work, we should have xfail test for it. 
But I think there must be a way to make it work :)